jsonrs/json.go

```diff
@@ -0,0 +1,135 @@
+package jsonrs
+
+import (
+	"io"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+)
+
+const (
+	// StdLib is the JSON implementation of package/json.
+	StdLib = "std"
+	// SonnetLib is the JSON implementation of github.com/rudderlabs/sonnet.
+	SonnetLib = "sonnet"
+	// JsoniterLib is the JSON implementation of github.com/json-iterator/go.
+	JsoniterLib = "jsoniter"
+	// DefaultLib is the default JSON implementation.
+	DefaultLib = SonnetLib
+)
+
+var Default = New(config.Default)
```

Comment:
@mihir20 @atzoum @lvrach do we need really this?

Wouldn't it be better if we just move away from global variables and promote Dependency Injection? We could then even remove all the global functions. Devs can simply build their own instance of this object and pass it around. Wdyt?